src/main/java/codeu/model/data/Event.java

```diff
@@ -176,12 +215,76 @@ public Event(User user) {
     messageCreationTime = null;
     authorIdForMessage = null;
     conversationTitleOfMessage = null;
+    conversationIdForMessage = null;
     messageContent = null;
     messageId = null;
     userCreationTime = user.getCreationTime();
     nameOfUser = user.getName();
     userId = user.getId();
-	}
+
+    mediaId = null;
+    ownermediaId = null;
+    mediaCreationTime = null;
+    titleOfMedia = null;
+    contentOfMedia = null;
+    contentTypeOfMedia = null;
+    conversationIdForMedia = null;
+  }
+
+  /**
+   * Constructs an new event using a Media object.
+   *
+   * @param media an object of type media
+   */
+  public Event(Media media) {
+    eventType = EventType.MEDIA;
+
+    userStore = UserStore.getInstance();
+    conversationStore = ConversationStore.getInstance();
+
+    titleOfConversation = null;
+    authorIdForConversation = null;
+    conversationCreationTime = null;
+    conversationId = null;
+
+    messageCreationTime = null;
+    authorIdForMessage = null;
+    conversationTitleOfMessage = null;
+    conversationIdForMessage = null;
+    messageContent = null;
+    messageId = null;
+    userCreationTime = null;
+    nameOfUser = null;
+    userId = null;
+
+    mediaId = media.getId();
+    ownermediaId = media.getOwnerId();
+    mediaCreationTime = media.getCreationTime();
+    titleOfMedia = media.getTitle();
+    contentOfMedia = media.getContent();
+    contentTypeOfMedia = media.getContentType();
+    conversationIdForMedia = media.getConversationId();
+  }
+
+  public UUID getMediaId() {
+    return mediaId;
+  }
+
+  public UUID getConversationIdForMedia() {
+    return conversationIdForMedia;
+  }
+
+  public UUID getConversationIdForMessage() {
+    return conversationIdForMessage;
+  }
+
+  public String getMessageContent() {
+    return messageContent;
+  }
+
+  public UUID getAuthorIdForMessage() {
+    return authorIdForMessage;
+  }
 
   public String getNameOfUser() {
```

Comment:
This getNameOfUser() method is really confusing to me:
1) Is it intended to return name of the user, hence NOT apply to any other event type? If this is the case the design I proposed earlier with Event being a parent class and NewUserEvent being a sub-class would be better, because getNameOfUser() would ONLY be a method of NewUserEvent, not any other sub-class like MediaUploadEvent()
2) There are MULTIPLE if(eventType == EventType.Media) in this function. I don't understand why. Any of the the "if(eventType == EventType.Media)" block (except the first one) will NEVER be executed.
3) There are places you are returning the epochSecond / converstaionId / mediaId, but the function is called getNameOfUser. these are NOT name of a user.

Reply:
1 is correct and therefore on a future iteration, I will try to optimize the code by creating an abstract class.